Sequences/MRML/vtkMRMLVolumeSequenceStorageNode.cxx

```diff
@@ -134,27 +131,17 @@ int vtkMRMLVolumeSequenceStorageNode::ReadDataInternal(vtkMRMLNode* refNode)
   const char* sequenceAxisUnit = reader->GetAxisUnit(0);
   volSequenceNode->SetIndexUnit(sequenceAxisUnit ? sequenceAxisUnit : "");
 
-  // Copy image data to sequence of volume nodes
-  vtkImageData* imageData = reader->GetOutput();
-  if (imageData == NULL || imageData->GetPointData()==NULL || imageData->GetPointData()->GetScalars() == NULL)
-  {
-    vtkErrorMacro("vtkMRMLVolumeSequenceStorageNode::ReadDataInternal: invalid image data");
-    return 0;
-  }
-  int numberOfFrames = imageData->GetNumberOfScalarComponents();
-  vtkNew<vtkImageExtractComponents> extractComponents;
-  extractComponents->SetInputConnection(reader->GetOutputPort());
-  for (int frameIndex = 0; frameIndex<numberOfFrames; frameIndex++)
+  // Read and copy the data to sequence of volume nodes
+  int numberOfFrames = reader->GetNumberOfSequenceFrames();
+  for (int frameIndex = 0; frameIndex < numberOfFrames; ++frameIndex)
   {
-    extractComponents->SetComponents(frameIndex);
-    extractComponents->Update();
-    vtkNew<vtkImageData> frameVoxels;
-    frameVoxels->DeepCopy(extractComponents->GetOutput());
-    // Slicer expects normalized image position and spacing
-    frameVoxels->SetSpacing( 1, 1, 1 );
-    frameVoxels->SetOrigin( 0, 0, 0 );
+    reader->SetSequenceFrameIndex(frameIndex);
+    reader->Update();
+    vtkImageData *imageData = reader->GetOutput();
+    imageData->SetOrigin(0, 0, 0);
+    imageData->SetSpacing(1, 1, 1);
     vtkNew<vtkMRMLScalarVolumeNode> frameVolume;
-    frameVolume->SetAndObserveImageData(frameVoxels.GetPointer());
+    frameVolume->SetAndObserveImageData(imageData);
```

Comment:
I would expect that the reader does not reallocate output vtkImageData when we change the current frame index and update. This means that we need to deep-copy of the reader output. Why it is not needed? Or maybe a deep copy is made when we set the image data in the sequence?

Reply:
I moved this PR to https://github.com/ProxSim/Sequences/pull/1. I'll reply to your comment there. Then I'll reopen this PR after we review there. Sorry for the mess, Thanks!